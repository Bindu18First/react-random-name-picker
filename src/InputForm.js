import React from "react";
import App from "./App"

class InputForm extends App {
  constructor() {
    super();
    this.state = {
      names: ""
    };
  }
  render(){
      console.log('in Inpurform',props)
      return(
            <div>
                Enter name:
                <input type="text" value={this.state.names}/>
            </div>
      )
  }
}
export default InputForm;
