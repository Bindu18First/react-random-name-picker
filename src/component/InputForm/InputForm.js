import React, { Component } from "react";

class InputForm extends Component {
  constructor() {
    super();
    this.state = {
      inputNames: ""
    };
  }
  handleChange(event) {
    this.setState({
      inputNames: event.target.value
    });
  }
  handleSubmit = event => {
    event.preventDefault();
    const newName = this.state.inputNames;
    this.props.addName(newName);
    this.setState({
      inputNames: ""
    });
  };
  render() {
    return (
      <div>
        <form onSubmit={event => this.handleSubmit(event)}>
          Enter name:
          <input
            type="text"
            value={this.state.inputNames}
            onChange={event => this.handleChange(event)}
          />
          <button>Add more Names</button>
        </form>
      </div>
    );
  }
}
export default InputForm;
