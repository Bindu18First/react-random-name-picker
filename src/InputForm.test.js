import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import InputForm from "./InputForm";

it("renders div with some snapshot", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<InputForm />);
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
