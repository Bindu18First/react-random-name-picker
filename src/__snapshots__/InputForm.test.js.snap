// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders div with some snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    Enter name:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button>
      Add more Names
    </button>
  </form>
</div>
`;
